docs/releases/3.0.txt

```diff
@@ -276,6 +276,19 @@ Django 3.0, we're removing these APIs at this time.
 
 * ``django.utils.safestring.SafeBytes`` - Unused since Django 2.0.
 
+Changed default for ``FILE_UPLOAD_PERMISSION`` setting to ``0o644``
```

Comment:
Suggested wording:
```
New default value for the ``FILE_UPLOAD_PERMISSIONS`` setting
-------------------------------------------------------------

In older versions, the :setting:`FILE_UPLOAD_PERMISSIONS` setting defaults to
``None``. With the default :setting:`FILE_UPLOAD_HANDLERS`, this results in
uploaded files having different permissions depending on their size and which
upload handler is used.

``FILE_UPLOAD_PERMISSION`` now defaults to ``0o644`` (the default mode for new
files on on most Linux systems) to avoid this inconsistency.
```
(is "the default mode for new files on on most Linux systems" the correct rationale?)

Reply:
I agree with your wording suggestion mostly.

As for the rationale, I think most Linux system has default `0o644` mode for files because permissions represented by this mode are actually reasonable choices. Owner of the file is able to read and write the file while execution is not permitted. Group and others can only read the file thereby restricting any modification.

This is a reasonable choice for user uploaded files as well. Generally, We don't want our web server or any other program to interpret user uploaded files since these files are untrusted.

What I want to derive from this is `0o644` is actually a good mode for such files even if the underlying system defaults to something else. We could just modify the last line as
`FILE_UPLOAD_PERMISSION now defaults to 0o644 to avoid this inconsistency.`

There is a minor difference between what you have suggested and from a practical point, there is no real difference on most systems. However, what I want to express is that we are setting `0o644` because it is a good choice not because it is a value most system defaults to.

Reply:
`...which is probably as restrictive as we can make it without leading to lots of people having all sorts of issues with their `www` user not having the right group permissions...` — but I'm not sure how well that reads. 😀

(i.e. you might go for `0o640` or something but I don't think that would be the best default as people struggle to set groups etc exactly how they should.) 

So maybe `...allowing read access for e.g. your web server process...`. We could link to the deployment checklist at that point.